uart/lcd_uart/README.adoc

```diff
@@ -0,0 +1,44 @@
+= Attaching a 16x2 LCD via TTL
+
+This example code shows how to interface the Raspberry Pi Pico to one of the very common 16x2 LCD character displays. Due to the large number of pins these displays use, they are commonly used with extra drivers or backpacks. In this example, we will use an Adafruit LCD display backpack, which supports communication over USB or TTL. An RGB display is also used, but the backpack is compatible with white/blue or one color displays too.
+
+The backpack processes a set of commands that are documented https://learn.adafruit.com/usb-plus-serial-backpack/command-reference[here] and preceded by the "special" byte 0xFE. The backpack does the ASCII character conversion and even supports custom character creation. In this example, we use the primary UART (uart0) to read characters and print them onto the LCD. We also define a special startup sequence and vary the display's backlight color.
+
+[NOTE]
+======
+These LCD displays can also be driven directly using GPIO without the use of an adapter board. That is beyond the scope of this example and not very commonly done.
+======
+
+TIP: The Adafruit backpack uses a microcontroller (At90USB162) of its own to interface with commands sent to it and the raw pin outputs, providing a convenient layer of abstraction over the LCD. The backpack breaks out a few pins from this microcontroller and the RX pin too, so you could program it to do more! The software flashed onto the board is also https://learn.adafruit.com/usb-plus-serial-backpack/downloads[available to inspect].
+
+WARNING: While this display will work at 3.3V, it will be quite dim. If powering the Pico via USB (which provides 5V) consider powering the backpack via VBUS (pin 40). However, if you do this, DO NOT connect any other pins on the backpack to the Pico (which is not 5V tolerant) apart from the needed TX (Pico) -> RX (backpack) as the backpack's IO are now at 5V.
+
+== Wiring information
+
+Wiring up the device requires 3 jumpers, to connect VCC (3.3v), GND, TX. The example here uses both of the Pico's UARTs, one (uart0) for stdio and the other (uart1)for communication with the backpack. Pin 8 is used as the TX pin. Power is supplied from the 3.3V pin.
```

Comment:
I dunno how you use both UARTs with just 3 jumper wires?

Reply:
We only need TX on the Pico's uart1 and the other is taken care of by however the user is connecting the Pico to their laptop. Either USB or via the TX/RX pins. But that's not specific to this example though, so I haven't mentioned that as the user can choose how to set up their stdio UART.

Reply:
It might be worth adding the "uart selection template" e.g.
```
# enable usb output, disable uart output
pico_enable_stdio_usb(lcd_uart 1)
pico_enable_stdio_uart(lcd_uart 0)
```
to the `CMakeLists.txt` ? (see section 4.1. of https://datasheets.raspberrypi.org/pico/getting-started-with-pico.pdf )

Reply:
Done, and also added an explanatory comment